### ISSUE TITLE
Problem of reproduction

### ISSUE THREAD

**Issue by YaooXu** [XiaoxinHe/G-Retriever#3]
Thanks for your great work!
Howeve, I have some problems in reproducing experiments on the WebQSP dataset. I use the same command in the readme to process the data and train the model, but the test acc is only 0.605, which is much lower than 0.6732 ± 0.0076 reported in the paper.
My command is as follows:
```bash
python train.py --dataset webqsp --model_name graph_llm
```
The results are as as follows:
![image](https://github.com/XiaoxinHe/G-Retriever/assets/44971199/fe3c62ee-8d41-449b-80ac-fb80757779f0)
The val loss can train loss are as follows:
![image](https://github.com/XiaoxinHe/G-Retriever/assets/44971199/5644d4d4-ba0c-48a7-bf4e-57b8875dfdce)

One possible reason for this could be related to cached graphs. The paper states that the average number of tokens in the WebQSP is 18, whereas in my cached graphs, it is only 8.1. Is there a potential issue with this difference?
original graphs
![image](https://github.com/XiaoxinHe/G-Retriever/assets/44971199/92337056-a7c0-4cf7-8846-517add4d63e7)

cached graphs
![image](https://github.com/XiaoxinHe/G-Retriever/assets/44971199/aecc3c9e-61d7-44bc-a69f-050efd557fdb)

**Comment by XiaoxinHe** [comment]
Hi,

Thank you for your interest in our work.

To reproduce our results, could you please uncomment the line in the retrieval.py file for the WebQSP dataset at the following URL? https://github.com/XiaoxinHe/G-Retriever/blob/ddff58cab0598967e0c47300a81bdc8e66033405/src/dataset/utils/retrieval.py#L41

The reason is that, in the WebQSP dataset, there are instances where many edges have the same attributes. This situation dilutes the prize assigned to the edges. In such cases, we need to adjust the cost of edges to ensure that at least one edge is chosen. For instance, there could be more than 50 edges with the same attribute "film.film.genre", which ranks first among the top-5 edges. A prize of 5 would be diluted to 0.1 (5/50). If the cost of an edge is 0.5, then no edges will be selected, and only a single node with the highest prize will be returned, which is undesirable.

Therefore, to reproduce our result, please remove the comment on this line. The updated code should yield a mean number of nodes of 15.06. The expected output of the experiment should resemble the following screenshot:
![Screenshot 2024-03-11 at 19 44 29](https://github.com/XiaoxinHe/G-Retriever/assets/48676630/c3ae96df-86aa-467d-b006-db9fbf6e8cae)

I will adjust the code later to make manual modification unnecessary. Thanks for your question.

**Comment by YaooXu** [comment]
Thanks for your timely reply! The performance improves after using the newest code and remove the comment (from 60.5 to 65.2), but there is still a little gap between my results and that in the paper (65.2 vs 67.3), which might caused by pytorch and cuda version.
